adapter/slack.py

```diff
@@ -28,6 +29,24 @@ def __init__(
     async def handle_events(self, req: Request):  # pragma: no cover
         return await self.handler.handle(req)
 
+    def send_generated_response(
+        self, channel: str, ts: str, engine: ChatEngine, question: str
+    ):
+        try:
+            self.logger.info("generating response")
+
+            chatbot_response = engine.generate_response(query=question)
+
+            self.logger.info("sending generated response")
+
+            self.app.client.chat_update(channel=channel, ts=ts, text=chatbot_response)
+
+        except ChatResponseGenerationError as e:  # pragma: no cover
+            self.logger.error(e)
+            return {
+                "text": "Something went wrong when trying to generate your response."
+            }

```

Comment:
When the error occurs, does returning this value actually will be sent as a slack message?

Reply:
Ohh I see. I might've missed that since this is outside of the original thread. If it was inside of it, it would normally be a Slack message. Thanks for pointing it out

Reply:
The fix has now been pushed